packages/tests/src/ui-test/localdebug/localdebug-command-and-response.test.ts

```diff
@@ -50,80 +88,126 @@ describe("Command And Response Bot Local Debug Tests", function () {
+        await driver.sleep(Timeout.startdebugging);
+        try {
           await waitForTerminal(
             LocalDebugTaskLabel.StartBotApp,
             LocalDebugTaskInfo.StartBotAppInfo
           );
         } catch {
           const dialog = new ModalDialog();
-          console.log(`click "Debug Anyway" button for error dialog`);
-          await dialog.pushButton("Debug Anyway");
-          console.log(`Clicked button "Debug Anyway"`);
+          console.log(`click "Cancel" button for error dialog`);
+          await dialog.pushButton("Cancel");
           await driver.sleep(Timeout.shortTimeLoading);
-          await waitForTerminal(
-            LocalDebugTaskLabel.StartBotApp,
-            LocalDebugTaskInfo.StartBotAppInfo
+          console.log(
+            `Clicked button "Cancel" for failing to attach to main target`
           );
+          await stopDebugging();
+          await driver.sleep(Timeout.stopdebugging);
+          try {
+            await killPort(3978);
+            console.log(`close port 3978 successfully`);
+          } catch (error) {
+            console.log(`close port 3978 failed`);
+          }
+          await startDebugging(DebugItemSelect.DebugInTeamsUsingChrome);
+          try {
+            await waitForTerminal(
+              LocalDebugTaskLabel.StartBotApp,
+              LocalDebugTaskInfo.StartBotAppInfo
+            );
+
+            // check if there is error "Debug Anyway"
+            await driver.sleep(Timeout.startdebugging);
+            await waitForTerminal(
+              LocalDebugTaskLabel.StartBotApp,
+              LocalDebugTaskInfo.StartBotAppInfo
+            );
+          } catch {
+            const dialog = new ModalDialog();
+            console.log(`click "Debug Anyway" button for error dialog`);
+            await dialog.pushButton("Debug Anyway");
+            console.log(`Clicked button "Debug Anyway"`);
+            await driver.sleep(Timeout.shortTimeLoading);
+            await waitForTerminal(
+              LocalDebugTaskLabel.StartBotApp,
+              LocalDebugTaskInfo.StartBotAppInfo
+            );
+          }
         }
-      }
 
-      const teamsAppId = await localDebugTestContext.getTeamsAppId();
-      const page = await initPage(
-        localDebugTestContext.context!,
-        teamsAppId,
-        Env.username,
-        Env.password
-      );
-      await validateBot(page, {
-        botCommand: "helloWorld",
-        expected: "Your Hello World App is Running",
-      });
+        const teamsAppId = await localDebugTestContext.getTeamsAppId();
+        const page = await initPage(
+          localDebugTestContext.context!,
+          teamsAppId,
+          Env.username,
+          Env.password
+        );
+        await validateBot(page, {
+          botCommand: "helloWorld",
+          expected: "Your Hello World App is Running",
+        });
+
+        // cli preview
+        console.log("======= debug with cli ========");
```

Comment:
can we extract these codes into a common method? just pass the env, folderpath.. ?

Reply:
i'm not sure how many these codes will be reused?

Reply:
cool... I took some of the common code and wrapped it into a reusable function. basically happy path case will add this step.